extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/RestrictedGuacamoleTunnelService.java

```diff
@@ -183,6 +185,19 @@ protected ModeledConnection acquire(RemoteAuthenticatedUser user,
             @Override
             public int compare(ModeledConnection a, ModeledConnection b) {
 
+                Directory<ConnectionGroup> directory;
+                ConnectionGroup group;
+				try {
+	            	// Get connection group directory
+					directory = a.getCurrentUser().getAuthenticationProvider().getUserContext(a.getCurrentUser()).getConnectionGroupDirectory();
+	                // Get authorized connection group
+	                group = directory.get(a.getParentIdentifier());
+                if (group.getType() == ConnectionGroup.Type.BALANCING)
+                	return getIdleConnections(a)
```

Comment:
Magic changes in expected behavior really need to be documented. The comparator here is earlier defined as "Sort connections in ascending order of usage", which is being (intentionally) violated here. To avoid confusion, and for the sake of maintainability, that behavior needs to be clearly defined just as it previously was.